tests/functional/splinter_test.c

```diff
@@ -866,248 +1004,314 @@ test_splinter_perf(trunk_config    *cfg,
       cache_reset_stats(spl->cc);
       // trunk_print(spl);
    }
+   platform_log("%s() completed.\n", __FUNCTION__);
+   return rc;
+}
 
-   ZERO_CONTENTS_N(curr_op, num_tables);
-   if (num_lookup_threads != 0) {
-      start_time = platform_get_timestamp();
+/*
+ * -----------------------------------------------------------------------------
+ * splinter_perf_lookups() --
+ *
+ * Work-horse routine to run n-threads to exercise lookups performance.
+ * -----------------------------------------------------------------------------
+ */
+static platform_status
+splinter_perf_lookups(platform_heap_id             hid,
+                      trunk_config                *cfg,
+                      test_config                 *test_cfg,
+                      trunk_handle               **spl_tables,
+                      task_system                 *ts,
+                      test_splinter_thread_params *params,
+                      uint64                       num_lookup_threads,
+                      uint8                        num_tables,
+                      uint32                       max_async_inflight,
+                      uint64                       total_inserts)
+{
+   platform_log("%s() starting num_lookup_threads=%lu"
+                ", max_async_inflight=%d ...\n",
+                __FUNCTION__,
+                num_lookup_threads,
+                max_async_inflight);
+   uint64          start_time = platform_get_timestamp();
+   platform_status rc;
 
-      for (uint64 i = 0; i < num_lookup_threads; i++) {
-         platform_status ret;
+   do_n_async_ctxt_inits(
+      hid, num_lookup_threads, num_tables, max_async_inflight, cfg, params);
 
-         for (uint8 j = 0; j < num_tables; j++) {
-            async_ctxt_init(
-               hid, max_async_inflight, &params[i].async_lookup[j]);
-         }
-         ret = task_thread_create("lookup thread",
-                                  test_trunk_lookup_thread,
-                                  &params[i],
-                                  trunk_get_scratch_size(),
-                                  ts,
-                                  hid,
-                                  &params[i].thread);
-         if (!SUCCESS(ret)) {
-            return ret;
-         }
-      }
-      for (uint64 i = 0; i < num_lookup_threads; i++) {
-         platform_thread_join(params[i].thread);
-      }
+   rc = do_n_thread_creates("lookup_thread",
+                            num_lookup_threads,
+                            params,
+                            ts,
+                            hid,
+                            test_trunk_lookup_thread);
+   if (!SUCCESS(rc)) {
+      return rc;
+   }
+   for (uint64 i = 0; i < num_lookup_threads; i++) {
+      platform_thread_join(params[i].thread);
+   }
 
-      total_time = platform_timestamp_elapsed(start_time);
+   uint64 total_time = platform_timestamp_elapsed(start_time);
 
-      uint64    num_async_lookups       = 0;
```

Comment:
Deleted block is seen below on L1062 onwards. Used `MAX()` to simplify if-else construction to find Max.